valohai/internals/download.py

```diff
@@ -1,12 +1,50 @@
 import contextlib
 import os
 import tempfile
+from typing import Any
+
+from requests import Response
+from valohai.internals.utils import uri_to_filename, get_sha256_hash
 
 
 # TODO: This is close to valohai-local-run. Possibility to merge.
+def resolve_datum(datum_id: str) -> Any:
+    try:
+        from valohai_cli.api import request  # type: ignore
+    except ImportError as ie:
+        raise RuntimeError("Can't resolve datum without valohai-cli") from ie
+    resp: Response = request(url=f"/api/v0/data/{datum_id}", method="GET")
+    resp.raise_for_status()
+    return resp.json()
+
+
+def verify_datum(response: Any, input_folder_path: str) -> str:
+    datum_obj = response
+    filename = datum_obj["name"]
+    checksums = {
+        "md5": datum_obj["md5"],
+        "sha1": datum_obj["sha1"],
+        "sha256": datum_obj["sha256"],
+    }
+    file_path = os.path.join(input_folder_path, filename)
+    if filename in os.listdir(input_folder_path) and checksums[
+        "sha256"
+    ] == get_sha256_hash(file_path):
+        return file_path
```

Comment:
I think my comment from the last review went missing, but as said, this shouldn't use `os.listdir()` :)
```suggestion
    if os.path.isfile(file_path):
        if datum_obj["sha256"] == get_sha256_hash(file_path):
            return file_path
```

Reply:
Fixed